src/dtnconfig.rs

```diff
@@ -233,11 +233,21 @@ impl DtnConfig {
         self.workdir = cfg.workdir;
         self.db = cfg.db;
     }
-    // Helper function that adds discovery destinations to a config struct
-    pub fn add_destination(&mut self, destination: String) -> std::io::Result<()> {
-        let addr: SocketAddr = destination
-            .parse()
-            .expect("Error: Unable to parse the provided address into IP format");
+
+    /// Helper function that adds discovery destinations to a config struct
+    /// 
+    /// When provided with an IP address without port the default port 3003 is appended
+    pub fn add_destination(&mut self, destination: &mut String) -> std::io::Result<()> {
+        let addr: SocketAddr = if destination.parse::<SocketAddr>().is_err() {
+            destination.push_str(":3003");
```

Comment:
as this is the only variant where something is added a non-mutable string as a parameter should be fine.
here `destination` could be locally reassigned to `format!()` to generate a new string with the port. then the function is self-contained and without side effects for the calling parent.